src/state/features/modal/index.tsx

```diff
@@ -0,0 +1,22 @@
+import { createSlice, PayloadAction } from "@reduxjs/toolkit";
+
+interface ModalState {
+  modalOpen: boolean;
+}
+
+const initialState: ModalState = {
+  modalOpen: false,
+};
+
+const modalSlice = createSlice({
+  name: "modal",
+  initialState,
+  reducers: {
+    handleModal(state, action: PayloadAction<boolean>) {
+      state.modalOpen = action.payload;
```

Comment:
Perhaps you can call this function `toggleModal`, and avoid using the payload. Just revert the previous state

Reply:
It's cleaner, I agree. Fixed!